PKGBUILD

```diff
@@ -36,6 +35,7 @@ source=(
   https://cdn.kernel.org/pub/linux/kernel/v${pkgver%%.*}.x/${_srcname}.tar.{xz,sign}
   https://github.com/archlinux/linux/releases/download/$_srctag/linux-$_srctag.patch.zst{,.sig}
   config  # the main kernel config file
+  rust-toolchain
```

Comment:
I'm not sure if a hard-coded `rust-toolchain` does make sense anymore, since from kernel 6.12 on the latest stable version should work.

But for building out-of-tree Rust kernel modules we need to know which rust version was used to build the kernel. So we could just generate the `rust-toolchain` file form the available rustc compiler during the preparation.

Reply:
You would know a lot better about that than me :) I only saw that it failed to build for me, and wanted to fix it. Feel free to change it another way, if that is better

Reply:
I updated the branch with the changes how I think it should be. That way it should work for both people using the `rustc` / `rust-src` package and `rustup`.

People using `rustc` directly will probably need to rebuild the kernel on every upgrade of rust if they want out of tree modules to work.

Reply:
Works very well, apart from the `$src_dir` -> `$srcdir` change I highlighted.
Thank you!